### PR TITLE
A4A: Implement 'Resend invite' functionality on the Team members page.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request adds the 'Resend invite' capability to the Team members page.

Screenshot 2024-09-09 at 6 59 16 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/967

## Proposed Changes

* Add a **'Resend invite'** button to the Actions column, including the event handler and mutation hook.
* Additional: Fix hardcoded Pending state for user invites and missing handler for expired state.

## Why are these changes being made?

* This allows users to resend an invite in case of problems with the original invitation.

## Testing Instructions

* Use the A4A live link and go to the `/team` page.
* Create a member invite.
* Confirm you received the email. (Make sure you do not access it)
* In the Team member table, resend the invite.
* Confirm you received another email.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

File: client/a8c-for-agencies/data/team/use-resend-member-invite.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+interface APIResponse {
+	success: boolean;
+}
+
+function resendMemberInviteMutation(
+	params: resendMemberInviteParams,
+	agencyId?: number
+): Promise< APIResponse > {
+	if ( ! agencyId ) {
+		throw new Error( 'Agency ID is required to assign a license' );
```
Comment (tiagonoronha) [inline client/a8c-for-agencies/data/team/use-resend-member-invite.ts]: This should say `Agency ID is required to resend invite`

File: client/a8c-for-agencies/sections/team/hooks/use-handle-member-action.ts [hunk 2]
```diff
@@ -14,12 +15,42 @@ export default function useHandleMemberAction( { onRefetchList }: Props ) {
...
 
 	return useCallback(
 		( action: string, item: TeamMember, callback?: () => void ) => {
+			if ( action === 'resend-user-invite' ) {
+				resendMemberInvite(
+					{ id: item.id },
+					{
+						onSuccess: () => {
+							dispatch(
+								successNotice( translate( 'The invitation has been resend.' ), {
```
Comment (tiagonoronha) [inline client/a8c-for-agencies/sections/team/hooks/use-handle-member-action.ts]: Minor grammar issue: The invitation has been ~resend~ **resent**.

**Review by tiagonoronha — CHANGES_REQUESTED** [review]
Some minor changes to language needed but otherwise this works as expected!

File: client/a8c-for-agencies/data/team/use-resend-member-invite.ts (hunk 1 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/data/team/use-resend-member-invite.ts]: Good catch! I forgot to remove this after copy/pasting from a different hook.

File: client/a8c-for-agencies/sections/team/hooks/use-handle-member-action.ts (hunk 2 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/team/hooks/use-handle-member-action.ts]: Thanks, this has been fixed. 👍

**Review by tiagonoronha — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/16691047).

**Pull request merged**